### PR TITLE
Add normalizer for Texas arcgis data

### PR DESCRIPTION
**Pull request by forrestrice**
Resolves: https://github.com/CAVaccineInventory/vaccine-feed-ingest/issues/51

TODO: I wasn't able to find the source uri for the Texas data

### CODE REVIEW

**Comment by forrestrice** [comment]
The linter seems unhappy but I'm not quite clear on why or how to fix it. I tried `poetry run black .` but I get `pyenv: black: command not found` even though `poetry show` does show black in the output. Probably missing something simple.

**Comment by MoralCode** [comment]
it appears as though the linter wants you to split function parameters up onto separate lines. i.e. change `logging.error(f"Error parsing date update_date_str={update_date_str}, update_time_str={update_time_str}: {e}")` to
```
        logging.error(
            f"Error parsing date update_date_str={update_date_str}, update_time_str={update_time_str}: {e}"
        )
       return None
 ```
 
 There seemed to be a few examples of this in the linter output

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        inventory=_get_inventory(site_attributes),
+        access=None,
+        parent_organization=None,
+        links=None,
+        notes=None,
+        active=None,
+        source=schema.Source(
+            source="arcgis",
+            id=site_attributes["OBJECTID"],
+            fetched_from_uri="UNKNOWN",  # TODO where is this from? Hard to tell from fetch.yaml
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/tx/arcgis/normalize.py:192]: This should be `https://tdem.maps.arcgis.com/apps/webappviewer/index.html?id=3700a84845c5470cb0dc3ddace5c376b`. Sorry for the mystery URL - I forgot to include them in the issues for many `arcgis` normalizers.

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    if public_phone:
+        source_phone = re.sub("[^0-9]", "", public_phone)
+        if len(source_phone) == 11:
+            source_phone = source_phone[1:]
+        if len(source_phone) == 10:
+            phone = f"({source_phone[0:3]}) {source_phone[3:6]}-{source_phone[6:]}"
+            contacts.append(schema.Contact(phone=phone))
+        else:
+            logging.warning(f"Phone Number is unexpected size:{public_phone}")
+            contacts.append(schema.Contact(other=public_phone))
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/tx/arcgis/normalize.py:120]: Thank you for the clear logging! From looking at the output that falls through, I see ~6 cases where the first 10 digits are a phone number, and then it's followed by either an extension or another number. Could we parse the first 10 as a phone number in these cases?

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        return None
+
+
+def _get_contacts(site_attributes: dict) -> Optional[List[schema.Contact]]:
+    contacts = []
+    public_phone = site_attributes["PublicPhone"]
+    if public_phone:
+        source_phone = re.sub("[^0-9]", "", public_phone)
+        if len(source_phone) == 11:
+            source_phone = source_phone[1:]
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/tx/arcgis/normalize.py:114]: Is this case to handle phone numbers like `+1 (223) 445-6789`? If so -thank you! Could you be sure to follow the same output format as below (with the parens/space/dash)?

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        ),
+        schema.Vaccine(
+            vaccine="pfizer_2", supply_level=site_attributes["PFIZER_AVAILABLE2"]
+        ),
+        schema.Vaccine(
+            vaccine="moderna", supply_level=site_attributes["MODERNA_AVAILABLE"]
+        ),
+        schema.Vaccine(
+            vaccine="moderna_2", supply_level=site_attributes["MODERNA_AVAILABLE2"]
+        ),
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/tx/arcgis/normalize.py:141]: Could you update the vaccines to normalize to `pfizer`, `moderna`, or `janssen ` (ignore the `_2` if it's provided)?

Also - where might be a good place in our documentation for me to note this request (I know it's not written anywhere at the moment, and I don't want to surprise you with it in a PR review)?

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+    if not address:
+        return None
+    match = re.search(r"([\S]*),\s?TX", address, flags=re.IGNORECASE)
+    if match:
+        return match.group(1)
+    match = re.search(r"([\S]*)\s[0-9]{5}", address, flags=re.IGNORECASE)
+    return match.group(1) if match else None
+
+
+def _get_address(site_attributes: dict) -> Optional[schema.Address]:
```
Comment (eliblock) [inline vaccine_feed_ingest/runners/tx/arcgis/normalize.py:73]: Thank you for parsing the addresses and adding clear logs when something does not parse!

**Review by eliblock — COMMENTED** [review]
Thank you for writing this normalizer! I answered your question about the source URL (sorry for that not being shared ahead of time) and made a few other suggestions.

I'm excited for this data to be processed soon!

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py (hunk 4 above)
Comment (MoralCode) [reply vaccine_feed_ingest/runners/tx/arcgis/normalize.py:141]: a note near the relevant part of the normalizing/schema section of the wiki might be nice

Comment (eliblock) [reply vaccine_feed_ingest/runners/tx/arcgis/normalize.py:141]: [Added](https://github.com/CAVaccineInventory/vaccine-feed-ingest/wiki/Runner-pipeline-stages#normalize)!

Comment (forrestrice) [reply vaccine_feed_ingest/runners/tx/arcgis/normalize.py]: Ok I represented these as the sum (e.g. moderna + moderna_2). Let me know if it should be only the first doses.
https://github.com/CAVaccineInventory/vaccine-feed-ingest/pull/134/commits/1bf88f731d68818a3af3242079fa2f5c2ee6c292

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py (hunk 1 above)
Comment (forrestrice) [reply vaccine_feed_ingest/runners/tx/arcgis/normalize.py]: https://github.com/CAVaccineInventory/vaccine-feed-ingest/pull/134/commits/1bf88f731d68818a3af3242079fa2f5c2ee6c292

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py (hunk 2 above)
Comment (forrestrice) [reply vaccine_feed_ingest/runners/tx/arcgis/normalize.py]: Added a phone regex that seems to handle all the valid numeric phone numbers:
https://github.com/CAVaccineInventory/vaccine-feed-ingest/pull/134/commits/484ff2a5766214024e39b6707117e17211771c6a

File: vaccine_feed_ingest/runners/tx/arcgis/normalize.py (hunk 3 above)
Comment (forrestrice) [reply vaccine_feed_ingest/runners/tx/arcgis/normalize.py]: done: https://github.com/CAVaccineInventory/vaccine-feed-ingest/pull/134/commits/484ff2a5766214024e39b6707117e17211771c6a

**Comment by eliblock** [comment]
I just realized that this and #191 are both normalizing the same data. I'm sorry about that snafu in assignment.

Unless there is a better suggestion, I'll take a look (tonight or tomorrow) at combining the approaches from these two PRs, and then will push the result to one (and merge it)... closing the other.

I'm sorry that I did not notice this conflict in my first round of reviews over the weekend, and regardless of how this ends up happening, I'm excited to get this data on the map tomorrow!

**Comment by forrestrice** [comment]
@eliblock No worries on that; I realized I didn't leave a comment on the issue. Use and/or remove as you see fit, won't bother me as long as the data is flowing :)

**Comment by eliblock** [comment]
Hi! I've been holding off merging these two PRs because we've been adding significant validation to the schema, and wanted to make sure this normalizer took advantage of it. Now that we've landed that (see #278), this should be ready to work on again!

**Comment by eliblock** [comment]
Closing in favor of #563. I'm excited that this data is now moving!

**Pull request closed without merging**